apps/dashboard/src/contract-ui/tabs/claim-conditions/components/claim-conditions-form/hooks.ts

```diff
@@ -0,0 +1,237 @@
+      ]);
+      const snapshot = await fetchSnapshot(
+        condition.merkleRoot,
+        contractMetadata.merkle,
+        options.contract.client,
+      );
+      return {
+        ...condition,
+        startTime: new Date(Number(condition.startTimestamp * 1000n)),
+        currencyAddress: condition.currency,
+        price: condition.pricePerToken,
+        maxClaimableSupply: toUnlimited(condition.maxClaimableSupply),
+        currencyMetadata,
+        currentMintSupply: (
+          condition.maxClaimableSupply - condition.supplyClaimed
+        ).toString(),
+        maxClaimablePerWallet: toUnlimited(condition.quantityLimitPerWallet),
+        merkleRootHash: condition.merkleRoot,
+        metadata,
+        snapshot,
+      } satisfies CombinedClaimCondition;
+    }),
+  );
+}
+
+type PhaseInput = z.input<typeof LegacyClaimConditionInputSchema>;
+
+export function setClaimPhasesTx(
+  baseOptions: BaseTransactionOptions<Options>,
+  rawPhases: PhaseInput[],
+) {
+  const phases = rawPhases.map((phase) => {
+    return {
+      startTime: toDate(phase.startTime),
+      maxClaimableSupply: toBigInt(phase.maxClaimableSupply),
+      maxClaimablePerWallet: toBigInt(phase.maxClaimablePerWallet),
+      merkleRootHash: phase.merkleRootHash as string | undefined,
+      overrideList: phase.snapshot?.length
+        ? snapshotToOverrides(phase.snapshot)
+        : undefined,
+      price: toBigInt(phase.price)?.toString(),
+      currencyAddress: phase.currencyAddress,
+      metadata: phase.metadata,
+    } satisfies ERC20Ext.SetClaimConditionsParams["phases"][0];
+  });
+
+  switch (baseOptions.type) {
+    case "erc20":
+      return ERC20Ext.setClaimConditions({
+        contract: baseOptions.contract,
+        phases,
+      });
+    case "erc721":
+      return ERC721Ext.setClaimConditions({
+        contract: baseOptions.contract,
+        phases,
+      });
+    case "erc1155":
+      return ERC1155Ext.setClaimConditions({
+        contract: baseOptions.contract,
+        phases,
+        tokenId: baseOptions.tokenId,
+      });
+  }
+}
+
+// HELPERS
+function snapshotToOverrides(snapshot: PhaseInput["snapshot"]) {
+  if (!snapshot) {
+    return [];
+  }
+
+  return snapshot.map((entry) => {
+    if (typeof entry === "string") {
+      return {
+        address: entry,
+      } satisfies OverrideEntry;
+    }
+    return {
+      address: entry.address,
+      maxClaimable: toBigInt(entry.maxClaimable)?.toString(),
+      price: toBigInt(entry.price)?.toString(),
+      currencyAddress: entry.currencyAddress,
+    } satisfies OverrideEntry;
+  });
+}
+function toDate(timestamp: number | Date | undefined) {
+  if (timestamp === undefined) {
+    return undefined;
+  }
+  if (timestamp instanceof Date) {
+    return timestamp;
+  }
+  return new Date(timestamp);
+}
+function toBigInt(value: string | number | undefined) {
+  if (value === undefined) {
+    return undefined;
+  }
+  if (value === "unlimited") {
```

Comment:
i already handle this internally, you should not have to convert anything yourself